src/Ledger/Transaction.lagda

```diff
@@ -173,16 +171,15 @@ the transaction body are:
 
   scriptOuts : UTxO → UTxO
   scriptOuts utxo = filterᵐ (sp-∘ (to-sp isScriptAddr?)
-                             λ { (fst , addr , snd) → addr}) utxo
+                             λ { (_ , addr , _) → addr}) utxo
 
   txinsScript : ℙ TxIn → UTxO → ℙ TxIn
   txinsScript txins utxo = txins ∩ dom (proj₁ (scriptOuts utxo))
 
   lookupScriptHash : ScriptHash → Tx → Maybe Script
-  lookupScriptHash sh tx
-    = M.map (lookupMap m)
-    $ decToMaybe
-    $ sh ∈? mapˢ proj₁ (m ˢ)
+  lookupScriptHash sh tx = case sh ∈? mapˢ proj₁ (m ˢ) of λ where
+      (no _)  → nothing
+      (yes _) → just $ lookupᵐ m sh
```

Comment:
That's a good point! I think we could also change `ifᵈ_then_else_` so that it passes the proof along as a hidden argument:
```agda
ifᵈ_then_else_ : (P : Set a) → ⦃ Dec P ⦄ → ({P} → X) → ({¬ P} → X) → X
ifᵈ P then t else f with ¿ P ¿
... | yes p = t {p}
... | no ¬p = f {¬p}
```
And then we could do this:
```agda
ifᵈ sh ∈ mapˢ proj₁ (m ˢ)
  then just $ lookupᵐ m sh
  else nothing
```
This should work, since the tactic should find hidden arguments just fine, and they are implicitly qualified here.